scripts/index.js

```diff
@@ -0,0 +1,209 @@
+function getAgeValidity(value) {
+  const isNumber = !Number.isNaN(value) && typeof value === 'number';
+  return isNumber && Number(value) > 0;
+}
+
+const newsAllPromise = fetch(
+  'https://hacker-news.firebaseio.com/v0/topstories.json?print=pretty'
+);
+newsAllPromise
+  .then((responses) => responses.json())
+  .then((newsAllIds) => {
+    const newsIdsSlice = newsAllIds.slice(0, 5);
+    const newsPromisesFive = newsIdsSlice.map((id) =>
+      fetch(
+        `https://hacker-news.firebaseio.com/v0/item/${id}.json?print=pretty`
+      )
+    );
+    return Promise.all(newsPromisesFive);
+  })
+  .then((responses) =>
+    Promise.all(responses.map((response) => response.json()))
+  )
+  .then((newsData) =>
+    Promise.all([
+      Promise.resolve(newsData),
+      ...newsData.reduce(
+        (accumulator, { kids }) => [
+          ...accumulator,
+          ...kids
+            .slice(0, 4)
+            .map((id) =>
+              fetch(
+                `https://hacker-news.firebaseio.com/v0/item/${id}.json?print=pretty`
+              )
+            ),
+        ],
+        []
+      ),
+    ])
+  )
+  .then(([newsData, ...commentsResponses]) =>
+    Promise.all([
+      Promise.resolve(newsData),
+      ...commentsResponses.map((responses) => responses.json()),
+    ])
+  )
+  .then(([newsData, ...comments]) => {
+    const newsHtml = newsData.reduce((accom, newsOne) => {
+      const {
+        by,
+        descendants,
+        time: newsTime,
+        title: newsTitle,
+        score,
+        url,
+        kids,
+      } = newsOne;
+
+      const commentsHtml = comments
+        .filter(({ id }) => kids.slice(0, 4).includes(id))
+        .reduce(
+          (acc, { text, time }) => ` ${acc} 
+        <li class="generic-list__item">
+           <article class="comment">
+             <p class="comment__info">
+               <span>${by}</span>
+               <span>${time}</span>
+             </p>
+             <div>${text}</div>
+            </article>
+         </li>`,
+          ''
+        );
+      const newsUrl = url ? new URL(url) : {};
+      const { hostname = null } = newsUrl;
+      const heading = `<h2 class="generic-list__title">${newsTitle}</h2>`;
+      const headingWithLink = `<a href="${url}">${heading}</a>`;
+      const source = url && hostname ? `<a href="${url}">${hostname}</a>` : '';
+
+      return `${accom}
+        <li class="generic-list__item">
```

Comment:
by name generic-list is something generic and a list
so it should take care about list and it's items only, not about what is inside items
please decouple from content
```<li class="generic-list__item">
        <article class="news">
          <header class="news__header">
            ${hostname ? headingWithLink : heading}
            ${source}
          </header>
          <footer class="news__footer">
            <span>${descendants}</span> points by
            <span>${by}</span>
            <span>${time}</span> | <!-- timestamp to readable date -->
            <button class="show-comments">${score} comments</button>
          </footer>
        </article>```

Reply:
note, instead of divs, I'd suggest using semantic tags like header and footer

Reply:
show-comments is a separate class

Reply:
Fixed